src/components/request/Request.tsx

```diff
@@ -38,25 +33,24 @@ const Request = () => {
     parsedIdentifier,
   } = useReader(state);
 
-  useEffect(() => {
-    const requester = searchParams.get("requester")?.trim();
-    const identifier = searchParams.get("identifier");
-    const timestamp =
-      searchParams.get("timestamp") && Number(searchParams.get("timestamp"));
-    const ancillaryData = searchParams.get("ancillaryData");
-    const chainId =
-      searchParams.get("chainId") && Number(searchParams.get("chainId"));
+  const isByTransactionParams = useIsByTransactionParams();
+  const { request, error } = useRequestParams(isByTransactionParams);
 
-    if (requester && identifier && timestamp && ancillaryData && chainId) {
-      client.setActiveRequest({
-        requester,
-        identifier,
-        timestamp,
-        ancillaryData,
-        chainId,
-      });
+  useEffect(() => {
+    // TODO: would be nice to do something with the error here, like redirect to the homepage
+    if (!error && request) {
+      if (isByTransactionRequest(request)) {
```

Comment:
this call is ready to be added it, might as well do it in this pr, its just one line:
`client.setActiveRequestByTransaction(params: { chainId: number; transactionHash: string; eventIndex?: number })`